ArcGISRuntimeSDKQt_CppSamples/LocalServer/LocalServerFeatureLayer/LocalServerFeatureLayer.cpp

```diff
@@ -60,37 +61,73 @@ void LocalServerFeatureLayer::componentComplete()
   // Set map to map view
   m_mapView->setMap(m_map);
 
-  QString dataPath = QDir::homePath() + "/ArcGIS/Runtime/Data";
+  // Check for ArcGIS Pro map package files
+  QString fileName = "PointsofInterest.mpkx";
+  QString dataPath = QDir::homePath() + "/ArcGIS/Runtime/Data/mpkx/" + fileName;
+
+  // Check to see if map package exists
+  if (!QFileInfo::exists(dataPath) || !QFileInfo(dataPath).isFile())
+  {
+    qDebug() << "ArcGIS Pro .mpkx file not found at" << dataPath << "\nChecking for .mpk file";
+    fileName = "PointsofInterest.mpk";
+    dataPath = QDir::homePath() + "/ArcGIS/Runtime/Data/mpk/" + fileName;
+    if (!QFileInfo::exists(dataPath) || !QFileInfo(dataPath).isFile())
+      qDebug() << "File:" << dataPath << "not found";
+    else
+      qDebug() << "Using .mpk file from" << dataPath;
+  }
```

Comment:
@jared-2016 are you suggesting using a QMessageBox popup?
@ldanzinger what are the implications of supporting the .mpk going forward? Do we want to get rid of support for it, rather than having this check?

Reply:
We shouldn't try and fall back to mpk if mpkx does not exist - .mpk won't be supported by local server at all for 100.10 - you must use .mpkx